pkg/beacon/beacon.go

```diff
@@ -199,3 +194,84 @@ func Initialize(
 
 	return nil
 }
+
+// Before we start relay entry signing process we need to confirm the current
+// relay request start block on the chain. This is to avoid having the client
+// participating in an old relay request signing that has already completed
+// with the rest of the signing group member clients and the result has been
+// already published to the chain.
+//
+// Such situation may happen when the current client received multiple blocks
+// at once after a longer delay and in those blocks to relay request events
+// to the same signing group were emitted.
+//
+// The confirmation mechanism has built-in retries. We can retry in case of an
+// error but also when the expected request start block does not match the one
+// currently registered on the chain. Such situation may happen for Infura-like
+// setup when two or more chain clients are behind a load balancer and they do
+// not have their state in sync yet.
+func confirmCurrentRelayRequest(
+	expectedRequestStartBlock uint64,
+	chain relaychain.RelayEntryInterface,
+	onConfirmed func(),
+	maxRetries int,
+	delay time.Duration,
+) {
+	for i := 1; ; i++ {
+		currentRequestStartBlock, err := chain.CurrentRequestStartBlock()
+		if err != nil {
+			if i == maxRetries {
+				logger.Errorf(
+					"could not check current request start block: [%v]; "+
+						"giving up after [%v] retries",
+					err,
+					maxRetries,
+				)
+				return
+			}
+
+			logger.Warningf(
+				"could not check current request start block: [%v]; "+
+					"will retry after [%v]",
+				err,
+				delay,
+			)
+			time.Sleep(delay)
+			continue
+		}
+
+		current := currentRequestStartBlock.Uint64()
```

Comment:
Thoughts on adding more context to the `current` variable name?
```suggestion
		currentRequestStartBlock := currentRequestStartBlockBigInt.Uint64()
```